pkg/policy/l4.go

```diff
@@ -1144,8 +1143,8 @@ type L4PolicyMap interface {
 	IngressCoversContext(ctx *SearchContext) api.Decision
 	EgressCoversContext(ctx *SearchContext) api.Decision
 	ForEach(func(l4 *L4Filter) bool)
-	Equals(t *testing.T, bMap L4PolicyMap) bool
-	Diff(t *testing.T, expectedMap L4PolicyMap) string
+	Equals(bMap L4PolicyMap) bool
+	Diff(expectedMap L4PolicyMap) string
```

Comment:
Ditto above.
```suggestion
	TestingOnlyDiff(expectedMap L4PolicyMap) string
```

Reply:
Thank you for the feedback, @nathanjsweet!

I've renamed the functions as you suggested to TestingOnlyEquals and TestingOnlyDiff to preserve the intent of restricting their usage to testing scenarios, I have also added a note in endpoint.go above the NewTestEndpointWithState function the it should be used for testing purposes only since I also removed "testing" from this file. 
Please let me know if there's anything else you'd like me to address.